packages/editor/src/components/editor-canvas/index.js

```diff
@@ -0,0 +1,327 @@
+ * block inside it, recursing through any nesting levels,
+ * and return its attributes.
+ *
+ * @param {Array} blocks A list of blocks.
+ *
+ * @return {Object | undefined} The Post Content block.
+ */
+function getPostContentAttributes( blocks ) {
+	for ( let i = 0; i < blocks.length; i++ ) {
+		if ( blocks[ i ].name === 'core/post-content' ) {
+			return blocks[ i ].attributes;
+		}
+		if ( blocks[ i ].innerBlocks.length ) {
+			const nestedPostContent = getPostContentAttributes(
+				blocks[ i ].innerBlocks
+			);
+
+			if ( nestedPostContent ) {
+				return nestedPostContent;
+			}
+		}
+	}
+}
+
+function checkForPostContentAtRootLevel( blocks ) {
+	for ( let i = 0; i < blocks.length; i++ ) {
+		if ( blocks[ i ].name === 'core/post-content' ) {
+			return true;
+		}
+	}
+	return false;
+}
+
+export default function EditorCanvas( {
+	// Ideally as we unify post and site editors, we won't need these props.
+	autoFocus,
+	dropZoneElement,
+	className,
+	renderAppender,
+} ) {
+	const {
+		renderingMode,
+		postContentAttributes,
+		editedPostTemplate = {},
+		wrapperBlockName,
+		wrapperUniqueId,
+	} = useSelect( ( select ) => {
+		const {
+			getCurrentPostId,
+			getCurrentPostType,
+			getCurrentTemplateId,
+			getEditorSettings,
+			getRenderingMode,
+		} = select( editorStore );
+		const postTypeSlug = getCurrentPostType();
+		const _renderingMode = getRenderingMode();
+		let _wrapperBlockName;
+
+		if ( postTypeSlug === 'wp_block' ) {
+			_wrapperBlockName = 'core/block';
+		} else if ( ! _renderingMode === 'post-only' ) {
+			_wrapperBlockName = 'core/post-content';
+		}
+
+		const editorSettings = getEditorSettings();
+		const supportsTemplateMode = editorSettings.supportsTemplateMode;
+		const postType = select( coreStore ).getPostType( postTypeSlug );
+		const canEditTemplate = select( coreStore ).canUser(
+			'create',
+			'templates'
+		);
+		const currentTemplateId = getCurrentTemplateId();
+		const template = currentTemplateId
+			? select( coreStore ).getEditedEntityRecord(
+					'postType',
+					'wp_template',
+					currentTemplateId
+			  )
+			: undefined;
+
+		return {
+			renderingMode: _renderingMode,
+			postContentAttributes: getEditorSettings().postContentAttributes,
+			// Post template fetch returns a 404 on classic themes, which
+			// messes with e2e tests, so check it's a block theme first.
+			editedPostTemplate:
+				postType?.viewable && supportsTemplateMode && canEditTemplate
+					? template
+					: undefined,
+			wrapperBlockName: _wrapperBlockName,
+			wrapperUniqueId: getCurrentPostId(),
+		};
+	}, [] );
+	const { isCleanNewPost } = useSelect( editorStore );
+	const {
+		hasRootPaddingAwareAlignments,
+		themeHasDisabledLayoutStyles,
+		themeSupportsLayout,
+	} = useSelect( ( select ) => {
+		const _settings = select( blockEditorStore ).getSettings();
+		return {
+			themeHasDisabledLayoutStyles: _settings.disableLayoutStyles,
+			themeSupportsLayout: _settings.supportsLayout,
```

Comment:
Why do we have `disableLayoutStyles` and `supportsLayout`? Seems confusing, we should only have one config to disable "layout" support.

Reply:
`disableLayoutStyles` pulls from `add_theme_support( 'disable-layout-styles' );` for a fairly specific use case to still have the layout support active, but not output any styles for the layout. Themes that do this provide their own styling, using the layout classnames.

So it's a distinct feature from disabling the layout support.